tableaudocumentapi/workbook.py

```diff
@@ -95,12 +109,45 @@ def save_as(self, new_filename):
     # Private API.
     #
     ###########################################################################
-    def _prepare_datasources(self, xmlRoot):
+    @staticmethod
+    def _prepare_datasource_index(datasources):
+        retval = weakref.WeakValueDictionary()
+        for datasource in datasources:
+            retval[datasource.name] = datasource
+
+        return retval
+
+    @staticmethod
+    def _prepare_datasources(xml_root):
         datasources = []
 
         # loop through our datasources and append
-        for datasource in xmlRoot.find('datasources'):
+        for datasource in xml_root.find('datasources'):
             ds = Datasource(datasource)
             datasources.append(ds)
 
         return datasources
+
+    @staticmethod
+    def _prepare_worksheets(xml_root, ds_index):
+        worksheets = []
+        worksheets_element = xml_root.find('.//worksheets')
+        if worksheets_element is None:
+            return worksheets
+
+        for worksheet_element in worksheets_element:
+            worksheet_name = worksheet_element.attrib['name']
```

Comment:
Just checking, given our XML's usual crazyness, the name is actually the name? :)